Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/dialog/DialogFragmentTemplate.kt

```diff
@@ -0,0 +1,41 @@
+package de.rki.coronawarnapp.ui.dialog
+
+import android.app.Dialog
+import android.content.DialogInterface
+import android.os.Bundle
+import androidx.appcompat.app.AlertDialog
+import androidx.fragment.app.DialogFragment
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.util.ContextExtensions.getColorCompat
+
+class DialogFragmentTemplate(
+    private val cancelable: Boolean = true,
+    private val isDeleteDialog: Boolean = false,
+    private val dismissAction: () -> Unit = { },
+    private val materialDialog: MaterialAlertDialogBuilder? = null,
+    private val config: MaterialAlertDialogBuilder.() -> Unit
+) : DialogFragment() {
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        isCancelable = cancelable
+        return materialDialog?.create() ?: MaterialAlertDialogBuilder(requireContext()).apply(config).create()
```

Comment:
`config` need to be applied to `materialDialog?.create()`  too? just incase some other extension need to be added it can have also effect on the provided builder

Reply:
Wouldn't applying the config here override what's already configured in the `materialDialog`?
I was thinking we are either passing a fully configured `MaterialAlertDialogBuilder`, or we're passing the config for a `MaterialAlertDialogBuilder` and we apply it here.

Reply:
Neverming, I see what you mean now. I've applied the config to the `materialDIalog` as well. Very nice :D